pymagicc/io.py

```diff
@@ -401,12 +400,38 @@ def process_header(self, header):
             The metadata in the header.
         """
         metadata = {}
+        # assume we start in header in case we are looking at legacy file which
+        # doesn't have the '---- HEADER ----' line
+        in_header = True
+        header_lines = []
         for line in header.split("\n"):
             line = line.strip()
-            for tag in self.header_tags:
-                tag_text = "{}:".format(tag)
-                if line.lower().startswith(tag_text):
-                    metadata[tag] = line[len(tag_text) + 1 :].strip()
+            if not line:
+                continue
+            if line == "---- HEADER ----":
+                in_header = True
+            elif line == "---- METADATA ----":
+                in_header = False
+            else:
+                if in_header:
+                    for tag in self.header_tags:
+                        tag_text = "{}:".format(tag)
+                        if line.lower().startswith(tag_text):
+                            metadata[tag] = line[len(tag_text) + 1 :].strip()
+                            break
+                    else:
+                        header_lines.append(line)
+                else:
+                    if ":" not in line:
+                        header_lines.append(line)
+                    else:
+                        bits = line.split(":")
+                        key = bits[0]
+                        value = ":".join(bits[1:])
+                        metadata[key.strip()] = value.strip()
+
+        if header_lines:
```

Comment:
So files with no header lines won't have a 'header' key in their metadata. Is that the intended functionality because now you always have to check if the key exists where as a blank header is effectively the same thing

Reply:
The thinking is this: if there’s a header there, use it. If there’s not, don’t add it so the user is forced to add one if/when they re-write. Alternately we could do what we do in `_MAGWriter` and autofill when writing if header isn’t there/is blank